CodeGeneration/Sources/SyntaxSupport/DeclNodes.swift

```diff
@@ -953,60 +999,83 @@ public let DECL_NODES: [Node] = [
     base: .syntaxCollection,
     nameForDiagnostics: "parameter list",
+    documentation: """
+      A list of function parameters that are type annotated and may have an optional label.
+      The function parameters are represented by `FunctionParameterListSyntax`.
+
+      ### Example
+
+      ```swift
+      func foo(bar: Int, baz: Int) {
+
+      }
+      ```
+      """,
     elementChoices: [.functionParameter]
   ),
 
   Node(
     kind: .functionParameter,
     base: .syntax,
     nameForDiagnostics: "parameter",
+    documentation: "A function parameter",
     parserFunction: "parseFunctionParameter",
     traits: ["WithTrailingComma", "WithAttributes", "WithModifiers"],
     children: [
       Child(
         name: "attributes",
         kind: .collection(kind: .attributeList, collectionElementName: "Attribute", defaultsToEmpty: true),
-        nameForDiagnostics: "attributes"
+        nameForDiagnostics: "attributes",
+        documentation: "Attributes that are attached to the parameter."
       ),
       Child(
         name: "modifiers",
         kind: .collection(kind: .declModifierList, collectionElementName: "Modifier", defaultsToEmpty: true),
-        nameForDiagnostics: "modifiers"
+        nameForDiagnostics: "modifiers",
+        documentation: "Modifiers that are attached to the parameter."
       ),
       Child(
         name: "firstName",
-        kind: .token(choices: [.token(.identifier), .token(.wildcard)])
+        kind: .token(choices: [.token(.identifier), .token(.wildcard)]),
+        documentation: "The label of this parameter that will be used when the function is called."
       ),
       Child(
         name: "secondName",
         kind: .token(choices: [.token(.identifier), .token(.wildcard)], requiresLeadingSpace: true),
         nameForDiagnostics: "internal name",
+        documentation:
+          "If this is specified, it is the name by which the parameter can be referenced inside the function body.",
         isOptional: true
       ),
       Child(
         name: "colon",
-        kind: .token(choices: [.token(.colon)])
+        kind: .token(choices: [.token(.colon)]),
+        documentation: "If the parameter has a label, the colon separating the label from the type."
       ),
       Child(
         name: "type",
         kind: .node(kind: .type),
-        nameForDiagnostics: "type"
+        nameForDiagnostics: "type",
+        documentation: "The parameter's type."
       ),
       Child(
         name: "ellipsis",
         kind: .token(choices: [.token(.ellipsis)]),
+        documentation: "If the parameter is variadic, `...` to indicate that.",
         isOptional: true
       ),
       Child(
         name: "defaultValue",
         deprecatedName: "defaultArgument",
         kind: .node(kind: .initializerClause),
         nameForDiagnostics: "default value",
+        documentation: "If the parameter has a default value, the function clause describing the default value.",
```

Comment:
I don’t think *function clause* is a term we use anywhere else. How about we just say it’s the expression, glossing over the fact that the `InitializerClauseSyntax` also contains the `=`. I think that should get the point across good enough.

```suggestion
        documentation: "If the parameter has a default value, the expression describing the default value.",
```